libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -120,6 +122,70 @@ void AP_DDS_Client::update_topic(sensor_msgs_msg_NavSatFix& msg, const uint8_t i
     msg.position_covariance[8] = vdopSq;
 }
 
+void AP_DDS_Client::update_topic(sensor_msgs_msg_BatteryState& msg, const uint8_t instance)
+{
+    if (instance >= AP_BATT_MONITOR_MAX_INSTANCES) {
+        return;
+    }
+
+    update_topic(msg.header.stamp);
+    auto &battery = AP::battery();
+    
+    if (!battery.healthy(instance))
+    {
+        msg.power_supply_status = 3; //POWER_SUPPLY_HEALTH_DEAD
+        msg.present = false;
+        return;
+    }
+    msg.present = true;
+
+    msg.voltage = battery.voltage(instance);
+    
+    float temperature; 
+    msg.temperature = (battery.get_temperature(temperature, instance)) ? temperature : NAN;
+    
+    float current;
+    msg.current = (battery.current_amps(current, instance)) ? -1 * current : NAN;
+    
+    float capacity = (float)battery.pack_capacity_mah(instance)/1000.0;
+    msg.capacity = capacity;
+
+    uint8_t percentage;
+    if (battery.capacity_remaining_pct(percentage, instance))
+    {
+        msg.percentage = percentage/100.0;
+        msg.charge = (percentage) * (capacity);
+    }
+    else
+    {
+        msg.percentage = NAN;
+        msg.charge = NAN;
+    }
+    
+    msg.design_capacity = NAN;
+ 
+    if (battery.current_amps(current, instance))
+    {
+        if (percentage == 100) {msg.power_supply_status = 4;} //POWER_SUPPLY_STATUS_FULL
+        else if (-1*current > 0.0) {msg.power_supply_status = 1;} //POWER_SUPPLY_STATUS_CHARGING
```

Comment:
Why not flip logic, removing an extra multiplication?
```suggestion
        else if (current > 0.0) {msg.power_supply_status = 1;} //POWER_SUPPLY_STATUS_CHARGING
```

Reply:
fair point, I thought it would be easier for people to read it at first, but I see how it isn't